weblate/addons/base.py

```diff
@@ -278,6 +278,16 @@ def component_update(self, component: Component) -> None:
         """Event handler for component update."""
         # To be implemented in a subclass
 
+    def get_addon_activity_logs(self) -> list:
+        """Return activity logs for add-on."""
+        from weblate.addons.models import AddonActivityLog
+
+        return list(
+            AddonActivityLog.objects.filter(
+                addon=self.instance,
+            ).order_by("-created")
+        )
```

Comment:
Avoid conversion to list here. This way you fetch all the results even if the query is paginated later:

```suggestion
    def get_addon_activity_logs(self) -> QuerySet[AddonActivityLog]:
        """Return activity logs for add-on."""
        return self.instance.addonactivitylog_set.order_by("-created")
```